### ISSUE TITLE
Reference leak for star dictionary updates

### ISSUE THREAD

**Issue by michieldwitte** [kayhayen/Nuitka#68]
Issue:
Memory leak

Tested on:
Windows 10
Python 3.6.1
Nuitka version 0.5.29.2 (pip) & 0.5.30rc3 (zip file from website)

Ubuntu 16.04.4 LTS (GNU/Linux 4.4.0-116-generic x86_64)
Python 3.5.2 
Nuitka version 0.5.28 (pip) & 0.5.29.2 (pip)

Code example:
```
while True:
    a = {
        'a': 1,
        'b': 2
        }
    b = {'c': 3, **a}
```

Steps to recreate:
nuitka example.py
run the example binary
(monitor the memory useage)

Extra info:
The problem is most likely being caused by the **a

**Comment by kayhayen** [comment]
Thanks for the on point report. I will look into this right now and consider it for a hotfix. I added a reference counting test without the loop (tests/basics/Referencing35.py I need to check if the syntax was allowed before, I never remember those things).

```
def simpleFunction9():
    a = {
        'a': 1,
        'b': 2
        }
    b = {'c': 3, **a}
    return b
```

And it says this:

> -simpleFunction9: PASSED
> +simpleFunction9: FAILED 127557 127568 leaked 11

So the helper function used to execute the dictionary update from a must be wrong and not release temporary variables or so. BTW: It's a shame that this is not statically optimized yet, but that is another issue.

**Comment by kayhayen** [comment]
I just pushed a fix to the factory branch, the helper function simply wasn't releasing its args. Adding that resolves the memory leak. I will release a hotfix with this contained soon.

**Comment by kayhayen** [comment]
This was released as part of 0.5.29.3
